src/components/menu/__internal__/submenu/submenu.component.tsx

```diff
@@ -393,6 +396,46 @@ const Submenu = React.forwardRef<
       }
     }, [characterString, submenuItemIds]);
 
+    useEffect(() => {
```

Comment:
comment: we could probably add some short circuiting here, for example if `segmentBlocks` is falsy there's no need to make any further queries as state will be set to false no matter what.